service/exporter/exporter.go

```diff
@@ -57,12 +20,52 @@ type namespace struct {
 	Name string
 	Pods []pod
 }
-type leanixExport struct {
+type Response struct {
 	Namespaces []namespace
 	LastUpdate time.Time
 }
 
-func getNamespaces(c *kubernetes.Clientset) []namespace {
+type Config struct {
+	Excludes []string
+}
+
+// New creates a new configured version service.
+func New(config Config) (*Service, error) {
+	// Settings.
+	if config.Excludes == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.Excludes must not be empty")
+	}
+	newService := &Service{
+		Config: config,
+	}
+
+	return newService, nil
+}
+
+// Service implements the version service interface.
+type Service struct {
+	Config
+}
+
+func (s *Service) Get(ctx context.Context) (*Response, error) {
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
```

Comment:
We use error masking in all of our microservices. This makes it easy to debug issues because you see some kind of stack trace when errors are logged. You can see how it is done in any of the microservices. See also https://github.com/giantswarm/giantswarm/wiki/go-style-guide#errors.